src/url-value.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TokenType, tokenize } from '@csstools/css-tokenizer'
+import {
+  isFunctionNode,
+  isTokenNode,
+  parseListOfComponentValues,
+} from '@csstools/css-parser-algorithms'
+
+export const urlValue = (string: string): string | undefined => {
+  const componentValues = parseListOfComponentValues(tokenize({ css: string }))
+  return componentValues
+    .map(componentValue => {
+      if (
+        isTokenNode(componentValue) &&
+        componentValue.value[0] === TokenType.URL
+      ) {
+        return componentValue.value[4].value
+      }
+
+      if (
+        isFunctionNode(componentValue) &&
+        componentValue.getName() === `url`
+      ) {
+        for (const token of componentValue.tokens()) {
+          if (token[0] === TokenType.String) {
+            return token[4].value
+          }
+        }
+      }
+
+      return undefined
+    })
+    .find(Boolean)
+}
```

Comment:
I think this is fine! I didn't try too hard to avoid bogus values in my implementation because we can generally assume that we'll receive well-formed input as a postcss-plugin. I mostly was catching errors/warnings for cases that would directly interfere with generating the fallback `@font-face`